src/site_upload/cache_api/cache_api.py

```diff
@@ -38,13 +38,19 @@ def cache_api_data(s3_client, s3_bucket_name: str, db: str, target: str) -> None
             "study": dp.split("__")[0],
             "name": dp.split("__")[1],
         }
-        versions = column_types[dp_detail["study"]][dp_detail["name"]]
+        studies = column_types.get(dp_detail["study"], {"name": None})
+        versions = studies.get(dp_detail["name"], None)
+        if versions is None:
+            print(f"{dp} not found in column_types")
+            continue
         for version in versions:
+            if version != dp:
+                continue
             dp_dict = {
                 **dp_detail,
                 **versions[version],
-                "version": version,
-                "id": f"{dp_detail['study']}__{dp_detail['name']}__{version}",
+                "version": version.split("__")[2],
+                "id": f"{dp_detail['study']}__{dp_detail['name']}__{version.split('__')[2]}",
```

Comment:
nit: Is `version` worth splitting separately into a variable, for clarity?

Like `_version_package, version_number = version.split("__", 2)` or something. (for example, I don't know what the first part of the version is, so I just made up a variable name)

Reply:
Yeah, I buy it